src/html.rs

```diff
@@ -51,34 +54,81 @@ pub fn simplify(html: &str, lang: &str) -> String {
             }
         }
 
-        for id in to_remove.drain(..) {
-            if let Some(mut node) = document.tree.get_mut(id) {
-                node.detach();
-            }
-        }
+        remove_ids(&mut document, to_remove.drain(..));
     } else {
         warn!("No sections to remove configured for lang {lang:?}");
     }
 
-    // Remove elements with no text that isn't whitespace.
+    // Handle remaining removals in single pass.
 
-    for element in document
+    for el in document
         .root_element()
         .descendants()
         .filter_map(ElementRef::wrap)
     {
-        if element.text().all(|t| t.trim().is_empty()) {
-            to_remove.push(element.id());
+        if is_image(&el) || is_empty_or_whitespace(&el) {
+            to_remove.push(el.id());
         }
     }
+    remove_ids(&mut document, to_remove.drain(..));
+
+    // Final transformations.
 
-    for id in to_remove.drain(..) {
+    let base_url = Url::parse(&format!("https://{}.wikipedia.org/wiki/", &lang)).unwrap();
+    fix_relative_urls(&mut document, base_url);
```

Comment:
Do we insert full URLs on every page now? That is an overhead, webview on iOS and Android should work properly with relative URLs. Let's investigate and fix it in a separate issue later (TODO may be good here too).

Reply:
When looking at this more I found that the html does include a [`base`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) element set to `//lang.wikipedia.org/wiki/`, but when opened as a file in firefox it assumes the scheme is `file:` so they don't work.
I'll remove this, and later if we run into a similar problem with the webviews, setting the scheme once in the base element should handle it.